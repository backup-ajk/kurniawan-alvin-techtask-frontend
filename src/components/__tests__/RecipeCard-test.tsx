import React from 'react';
import renderer from 'react-test-renderer';

import { RecipeCard } from '../';

it('should render Recipe Card successfully', () => {
  const recipe = {
    title: 'KFC Rice',
    ingredients: ['Rice', 'Original KFC', 'Water'],
  };
  const recipeCard = renderer.create(<RecipeCard recipe={recipe} />).toJSON();
  expect(recipeCard).toMatchSnapshot();
});

it('should display correct information', () => {
  const recipe = {
    title: 'KFC Rice',
    ingredients: ['Rice', 'Original KFC', 'Water'],
  };
  const rendered = renderer.create(<RecipeCard recipe={recipe} />);
  const list = rendered.root.findAllByType('li');

  expect(list.length).toEqual(recipe.ingredients.length);
});
