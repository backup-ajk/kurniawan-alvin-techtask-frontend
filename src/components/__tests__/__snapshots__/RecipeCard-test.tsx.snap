// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Recipe Card successfully 1`] = `
<div
  style={
    Object {
      "borderColor": "#546e7a",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 5,
      "cursor": "pointer",
      "display": "flex",
      "flexDirection": "column",
      "height": 200,
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#607d8b",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "color": "#fff",
        "display": "flex",
        "fontSize": 24,
        "fontWeight": 600,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    KFC Rice
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#cfd8dc",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "display": "flex",
        "flex": 1,
        "overflowY": "scroll",
      }
    }
  >
    <ul
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginTop": 5,
          "padding": 0,
          "width": "100%",
        }
      }
    >
      <li
        style={
          Object {
            "display": "block",
            "fontSize": 18,
            "fontWeight": 500,
            "marginBottom": -1,
            "padding": "0.5rem",
            "position": "relative",
          }
        }
      >
        Rice
      </li>
      <li
        style={
          Object {
            "display": "block",
            "fontSize": 18,
            "fontWeight": 500,
            "marginBottom": -1,
            "padding": "0.5rem",
            "position": "relative",
          }
        }
      >
        Original KFC
      </li>
      <li
        style={
          Object {
            "display": "block",
            "fontSize": 18,
            "fontWeight": 500,
            "marginBottom": -1,
            "padding": "0.5rem",
            "position": "relative",
          }
        }
      >
        Water
      </li>
    </ul>
  </div>
</div>
`;
